libs/SerialNumberKit/include/SerialNumberKit.h

```diff
@@ -5,21 +5,28 @@
 #pragma once
 
 #include <array>
+#include <filesystem>
 #include <interface/drivers/MCU.h>
+#include <span>
 
 namespace leka {
 
 class SerialNumberKit
 {
-	static constexpr size_t SN_SIZE		  = 33;
-	static constexpr size_t SHORT_SN_SIZE = 16 + 1;
+	static constexpr size_t SN_SIZE			= 31 + 1;
+	static constexpr size_t SHORT_SN_SIZE	= 15 + 1;
+	static constexpr auto date_of_test_path = "fs/sys/date_of_test";
```

Comment:
clang-format define it as `const char *`

And `static constexpr auto` part struggles with `std::filesystem::path`, but yes it is the best type to represent `date_of_test_path`

Reply:
> clang-format define it as const char *

because of `auto`, the compiler makes the choice. clang might use const char *, while gcc will use something else.

if it's not important, then we don't care, `auto` is good -- if we want to force something for reasons X and Y, then the type must be defined.

> And static constexpr auto part struggles with std::filesystem::path, but yes it is the best type to represent date_of_test_path

why do you need the `static constexpr auto`?

using auto for member variables is not *"really"* allowed, if we want static polymorphism, that's what templates are for.

having said that, you can just have `const/constexpr std::filesystem::path`, you don't need the `static`, or am I missing something?

Reply:
ok ok

Instead of just redefined this part to be compliant with guidelines, it seems more interesting to implement the system as you done for os_version as you mention in the conversation below